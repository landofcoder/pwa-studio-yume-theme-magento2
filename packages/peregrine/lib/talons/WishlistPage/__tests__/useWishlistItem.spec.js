import React, { useEffect } from 'react';
import { act } from 'react-test-renderer';
import { useMutation } from '@apollo/client';

import createTestInstance from '../../../util/createTestInstance';
import { useWishlistItem } from '../useWishlistItem';

jest.mock('@apollo/client', () => ({
    useMutation: jest.fn().mockReturnValue([jest.fn(), { loading: false }])
}));

jest.mock('../../../context/cart', () => {
    const state = { cartId: 'cart123' };
    const api = {};

    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

const log = jest.fn();
const Component = props => {
    const talonProps = useWishlistItem({ ...props });

    useEffect(() => {
        log(talonProps);
    }, [talonProps]);

    return null;
};

const baseProps = {
    mutations: {},
    sku: 'shoggoth-shirt'
};

test('it returns the correct shape', () => {
    createTestInstance(<Component {...baseProps} />);

    const talonProps = log.mock.calls[0][0];

    expect(talonProps).toMatchSnapshot();
});

test('it returns mutation response fields', () => {
    useMutation.mockReturnValue([jest.fn(), { error: {}, loading: true }]);
    createTestInstance(<Component {...baseProps} />);

    const talonProps = log.mock.calls[0][0];

    expect(talonProps.hasError).toBe(true);
    expect(talonProps.isLoading).toBe(true);
});

test('mutation passes options for simple item', () => {
    createTestInstance(<Component {...baseProps} />);

    const mutationOptions = useMutation.mock.calls[0][1];

    expect(mutationOptions).toMatchSnapshot();
});

test('mutation passes options for configurable item', () => {
    const configurableProps = {
        ...baseProps,
        childSku: 'shoggoth-shirt-xl-black'
    };
    createTestInstance(<Component {...configurableProps} />);

    const mutationOptions = useMutation.mock.calls[0][1];

    expect(mutationOptions).toMatchSnapshot();
});

test('handleAddToCart callback fires mutation', () => {
    const mockMutate = jest.fn();
    useMutation.mockReturnValue([mockMutate, { loading: false }]);
    createTestInstance(<Component {...baseProps} />);

    const talonProps = log.mock.calls[0][0];

    act(() => {
        talonProps.handleAddToCart();
    });

    expect(mockMutate).toHaveBeenCalled();
});
