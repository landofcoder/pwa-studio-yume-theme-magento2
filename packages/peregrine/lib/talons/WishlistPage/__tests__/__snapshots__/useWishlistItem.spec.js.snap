// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it returns the correct shape 1`] = `
Object {
  "confirmRemovalIsOpen": false,
  "handleAddToCart": [Function],
  "handleHideDialogs": [Function],
  "handleRemoveProductFromWishlist": [Function],
  "handleShowConfirmRemoval": [Function],
  "handleShowMoreActions": [Function],
  "hasError": false,
  "hasRemoveProductFromWishlistError": false,
  "isLoading": false,
  "isRemovalInProgress": false,
  "moreActionsIsOpen": false,
}
`;

exports[`mutation passes options for configurable item 1`] = `
Object {
  "variables": Object {
    "cartId": "cart123",
    "cartItem": Object {
      "data": Object {
        "quantity": 1,
        "sku": "shoggoth-shirt-xl-black",
      },
      "parent_sku": "shoggoth-shirt",
      "variant_sku": "shoggoth-shirt-xl-black",
    },
  },
}
`;

exports[`mutation passes options for simple item 1`] = `
Object {
  "variables": Object {
    "cartId": "cart123",
    "cartItem": Object {
      "data": Object {
        "quantity": 1,
        "sku": "shoggoth-shirt",
      },
    },
  },
}
`;
